x/auth/keeper/keeper.go

```diff
@@ -186,6 +187,21 @@ func (ak AccountKeeper) NextAccountNumber(ctx context.Context) uint64 {
 	if err != nil {
 		panic(err)
 	}
+	if n == 0 {
```

Comment:
it's technically consensus breaking. but this condition should not be true in production network.

Reply:
just think we only set new AccountNumber in old blocks which can't be done in migration